ttnctl/cmd/applications_pf_set.go

```diff
@@ -90,11 +90,14 @@ The functions are read from the supplied file or from STDIN.`,
 ########## Write your Validator here and end with Ctrl+D (EOF):`)
 				app.Validator = readFunction()
 			case "encoder":
-				fmt.Println(`function (val) {
-  // val is the output of the encoder function.
-
-  // todo: return an array of numbers
-  return return [96, 4, 3, 2, 1, 0, 1, 0, 1, 0, 0, 0, 0];
+				fmt.Println(`function encoder(jsonPayload) {
+	var str = JSON.stringify(jsonPayload);
+	strTab = Array.prototype.slice.apply(str);
```

Comment:
As @johanstokking already said stringify the full payload is probably not a best practice, but it doesn't seem to actually work either, or at least a test with this code fails in the dashboard with:

```js
Error("Could not dry-run downlink with fields on Handler: rpc error: code = 2 desc = Encoder not valid: should return an Array of integer numbers")
```

Which is weird because it does return an array of numbers.

Also using this version:

```js
function Encoder(obj) {
	var str = obj.greeting; // 'Hi'
	var bytePayload = [];
	var strTab = Array.prototype.slice.apply(str);
	for (i = 0; i < strTab.length; i++) {
		bytePayload.push(strTab[i].charCodeAt(0));
	}
	return bytePayload;
}
```

Or the more simple:

```js
function Encoder(obj) {
	var str = obj.greeting; // 'Hi'
	return str.split('').map(function(char) {
		return char.charCodeAt();
	});
}
```